### PR TITLE
polyfilling page visibility api

### PR DESCRIPTION
**Pull request by wheresrhys**
Difficult to write tests for
As these are read only properties d'you think I should do more to guard against errors?
@triblondon

### CODE REVIEW

File: polyfills/PageVisibility/polyfill-moz.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Page Visibility API
+(function () {
+	document.hidden = typeof document.hidden === 'undefined' ? document.mozHidden : document.hidden;
+	document.visibilityState = typeof document.visibilityState === 'undefined' ? document.mozVisibilityState : document.visibilityState;
```
Comment (triblondon) [inline polyfills/PageVisibility/polyfill-moz.js]: This feature detect is unnecessary - you can assume the native properties are undefined.

**Comment by triblondon** [comment]
Look fine in principle.  I'm phasing out variants, so can you combine these into a single polyfill that detects mozVisibilityState and webkitVisibilityState and acts accordingly?  Suspect you can remove much of the repetition.

**Comment by wheresrhys** [comment]
Amended as requested. It _will not_ pass the darn tests though. I'm unsure whether this is a problem with the pipeline, with my code or with my tests https://travis-ci.org/Financial-Times/polyfill-service/builds/97417131#L3229

**Comment by triblondon** [comment]
It's a problem with Sauce. I can't get to the bottom of it.

File: polyfills/PageVisibility/tests.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+describe('Page Visibility API', function () {
```
Comment (triblondon) [inline polyfills/PageVisibility/tests.js]: No need for this wrapper.

File: polyfills/PageVisibility/polyfill.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// Page Visibility API
+(function () {
+	var prefix = document.webkitVisibilityState ? 'webkit' : document.mozVisibilityState ? 'moz' : null;
+
+	if (!prefix) {
+		return;
+	}
+
+	document.visibilityState = document[prefix + 'VisibilityState'];
+	document.hidden = document[prefix + 'Hidden'];
```
Comment (triblondon) [inline polyfills/PageVisibility/polyfill.js]: I'd put this in a function, and call it from the event handler to avoid repeating the assignment code.

**Comment by triblondon** [comment]
I'll run the tests (which Sauce seems to be OK with from my local env).

File: polyfills/PageVisibility/config.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
+	"aliases": [
+		"default",
+		"caniuse:visib"
```
Comment (matthew-andrews) [inline polyfills/PageVisibility/config.json]: should this be `caniuse:pagevisibility` ?

File: polyfills/PageVisibility/config.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+	],
+	"browsers": {
+		"firefox": "10 - 17",
+		"safari": "4",
+		"android": "4.4 - *",
+		"opera": "15 - 19",
+		"chrome": "14 - 32"
```
Comment (matthew-andrews) [inline polyfills/PageVisibility/config.json]: also needed in BB7?

File: polyfills/PageVisibility/config.json (hunk 4 above)
Comment (wheresrhys) [inline polyfills/PageVisibility/config.json]: turns out both are supported

Comment (matthew-andrews) [inline polyfills/PageVisibility/config.json]: okidoke

**Comment by triblondon** [comment]
Test failure on Android 4.4

image

**Comment by wheresrhys** [comment]
That's a bit small - what does it say?

**Comment by triblondon** [comment]
CI passing now (Travis not working)

**Pull request merged**